src/main/java/com/squareup/kotlinpoet/FunSpec.kt

```diff
@@ -283,6 +301,17 @@ class FunSpec private constructor(builder: Builder) {
       parameters += parameterSpec
     }
 
+    fun delegateConstructor(constructor: Constructor, vararg parameters: String) =
+        delegateConstructor(constructor, *parameters.map { CodeBlock.of(it) }.toTypedArray())
+
+    fun delegateConstructor(
+        constructor: Constructor,
+        vararg parameters: CodeBlock = emptyArray()) = apply {
+      check(name.isConstructor) { "only constructors can have delegate parameters!" }
+      delegateConstructor = constructor
+      delegateConstructorParameters += parameters
```

Comment:
The `+=` call seems like it’s broken for repeated calls? Ie. it adds, rather than replaces?

Reply:
Indeed, good call. I guess it would make sense to store them in a set, will fix.

Reply:
On a closer look, using a set doesn't make sense, as you can pass a single `CodeBlock` as an argument for multiple parameters. In this case, appending to the list rather than replacing seems perfectly fine to me. Do we want API to allow replacing or removing the args?